pennylane/data/dataset.py

```diff
@@ -233,6 +253,8 @@ def __getattribute__(self, name):
                     if name not in value:
                         raise
                     value = value[name]
+                if name == "hamiltonian":
+                    value = self.__dict_to_hamiltonian(value["terms"], value["wire_map"])
```

Comment:
Nice, so the disk version of the dataset is kept the same then?

Reply:
correct. it's converted then saved to a Dataset whenever fetched, and that's it